libraries/lib-utility/MemoryX.cpp

```diff
@@ -0,0 +1,5 @@
+#include "MemoryX.h"
+
+// Make the symbol table non-empty
+UTILITY_API void lib_utility_dummy_symbol()
```

Comment:
Maybe we should add the concept of the header-only libraries to AudacityFunctions?

Reply:
I had the same thought.

However, see also my attempts to make the Mac C++17 build work without increasing the minimum SDK.  I did end up adding linked symbols to MemoryX.cpp after all.